lms/djangoapps/instructor/views/api.py

```diff
@@ -1092,6 +1092,28 @@ def get_coupon_codes(request, course_id):  # pylint: disable=unused-argument
     return instructor_analytics.csvs.create_csv_response('Coupons.csv', header, data_rows)
 
 
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
```

Comment:
We'll also have to assert that the end-user has the 'finance_admin' role in order to run this.